app/models/user.rb

```diff
@@ -1,10 +1,36 @@
 class User < ApplicationRecord
-  before_save {self.email = email.downcase}
+  attr_accessor :remember_token
+  before_save{self.email = email.downcase}
   validates :name, presence: true, length: {maximum: Settings.validate.name.maximum}
   VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
   validates :email, presence: true, length: {maximum: Settings.validate.email.maximum},
     format: {with: VALID_EMAIL_REGEX},
     uniqueness: {case_sensitive: false}
   has_secure_password
   validates :password, presence: true, length: {minimum: Settings.validate.password.minimum}
+
+  def remember
+    self.remember_token = User.new_token
+    update_attribute :remember_digest, User.digest(remember_token)
+  end
+
+  class << self
+    def User.digest string
```

Comment:
As you wrapped the class method inside the class << self block, no need to declare a class method using it class name. In summary remove **User** 
```ruby
def digest string
  # Method body
end 
```

Reply:
I forget to remove User.. thanks